Testscripts/Windows/NET-Boot-NoNIC-HotAdd-NIC.ps1

```diff
@@ -0,0 +1,243 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the Apache License.
+
+<#
+.Synopsis
+    Hot Add a NIC to a Gen2 VM that booted without a NIC.
+
+.Description
+    Test case for booting a Gen2 VM without a NIC, hot add a
+    NIC, verify it works, then hot remove the NIC.
+#>
+
+function Main {
+    param (
+        $VMName,
+        $HvServer,
+        $Ipv4,
+        $VMPort,
+        $VMUserName,
+        $VMPassword,
+        $RootDir
+    )
+
+    $KVP_KEY = "HotAddTest"
+    $Switch_Name = "External"
+
+    # Change the working directory to where we should be
+    Set-Location $RootDir
+
+    # Verify the target VM exists, and that it is a Gen 2 VM
+    "Info : Verify the SUT VM '${VMName}' exists"
+    $vm = Get-VM -Name $VMName -ComputerName $HvServer -ErrorAction SilentlyContinue
+    if (-not $?) {
+        LogErr "Unable to find VM '${VMName}' on server '${HvServer}'"
+    }
+
+    if ($vm.Generation -ne 2) {
+        LogWarn "This test requires a Gen 2 VM.  VM '${VMName}' is not a Gen2 VM"
```

Comment:
LogErr  and rephrase this better, required a gen2, is not gen2.
Keep consistent name "Gen2 VM"